src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -251,14 +255,20 @@ protected virtual SyntaxNode DefaultNodeExtractor(SyntaxNode node, ISyntaxFactsS
                 }
             }
 
-            // a = b;
-            // -> b
+            // `a = b;`
+            // -> `b`
             if (syntaxFacts.IsSimpleAssignmentStatement(node))
             {
                 syntaxFacts.GetPartsOfAssignmentExpressionOrStatement(node, out _, out _, out var rightSide);
                 return rightSide;
             }
 
+            // a => `b`; -> `a => b;`
+            if (syntaxFacts.IsLambdaBody(node))
+            {
+                return node.Parent;
+            }
```

Comment:
I'm not at all sure how i feel aobut this.  If someone selects the body of a lambda, i don't see how that makes me feel like they may have intended to select the lambda itself.   what's the use case?

Reply:
```csharp
    static void Main()
    {
        Action a = [||]() => {
            Console.WriteLine();
        };
    }
```

Reply:
We could also just ask for the body as well in the refactoring in question, that's another solution but I can't see any refactorings where we might not want the extraction, tbh.

Reply:
But that's not the body... so i'm not sure how this applies.  The body is the part after the `=>`

Reply:
Copied wrong test, my bad: 
```csharp
    static void Main()
    {
        Bar(x => [||]Goo(x));
    
```

Reply:
I agree with Cyrus here - we don’t really need to special case selection of body as mapping to the whole method node. If you select the method body, you only get refactorings that apply to method’s BlockSyntax or Expression body syntax. For your test, either the cursor needs to be just before ‘x’ or selection needs to contain the arrow token to get lambda specific refactorings.

Reply:
Will update the test in question then.